### PR TITLE
docs: Talk about sysroot, growing

### PR DESCRIPTION
**Pull request by cgwalters**
This one is also important to understand as growing the rootfs is a corner case.

### CODE REVIEW

File: docs/src/filesystem-sysroot.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Filesystem: Physical /sysroot
+
+The bootc project uses [ostree](https://github.com/ostreedev/ostree/) as a backend,
+and maps fetched container images to a "deployment": <https://ostreedev.github.io/ostree/deployment/>.
```
Comment (jeckersb) [inline docs/src/filesystem-sysroot.md:4]: `[deployment](https://ostreedev.github.io/ostree/deployment)` instead to make a pretty link?

File: docs/src/filesystem-sysroot.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+expect the `/` mount point to be the physical root.  Tools like
+this will need to be adjusted to instead detect this and operate
+on `/sysroot`.
+
+### Growing the block device
+
+Fundamentally bootc is agnostic to the underlying block device setup.
+How to grow the root block device depends on the underlying
+storage stack, from basic partitions to LVM.  However, a
+common tool is the https://manpages.debian.org/testing/cloud-guest-utils/growpart.1.en.html[growpart]
```
Comment (jeckersb) [inline docs/src/filesystem-sysroot.md:42]: bad link, should be - `[growpart](https://manpages.debian.org/testing/cloud-guest-utils/growpart.1.en.html)`

**Review by jeckersb — CHANGES_REQUESTED** [review]
Content looks good, just a couple nits on your favorite topic of trying to keep link syntax straight :laughing:

File: docs/src/filesystem-sysroot.md (hunk 2 above)
Comment (cgwalters) [reply docs/src/filesystem-sysroot.md]: Ah yes...hmm, I need to figure out if there's some markdown linter which can find this stuff.  The asciidoc vs markdown thing...argh

File: docs/src/filesystem.md [hunk 3]
```diff
@@ -16,6 +16,15 @@ enabled = true
...
 This will ensure that the entire `/` is a read-only filesystem.
 
+## Understanding physical vs logical root with `/sysroot`
+
+When the system is fully booted, it is into the equivalent of a `chroot`.
+The "physical" host root filesystem will be mounted at `/sysroot`.
+For more on this, see [filesystem: sysroot](filesystem-sysroot.md).
+
+All the remaining filesystem paths below are part of a "deployment root" which
+is used as a final target for the system boot.
```
Comment (jlebon) [inline docs/src/filesystem.md:26]: To connect this more with the previous paragraph, e.g.

```suggestion
This `chroot` filesystem is called the "deployment root". All the remaining
filesystem paths below are part of the deployment root which is used as a
final target for the system boot.
```

**Comment by jeckersb** [comment]
Hm I tried to use the github UI editor for the first time ever to add the little syntax fixup  but forgot about DCO signoff so now it's mad about that

**Review by jlebon — APPROVED** [review]

**Pull request merged**